Commands/Base/DisconnectSPOnline.cs

```diff
@@ -54,23 +54,23 @@ protected override void ProcessRecord()
 
         internal static bool DisconnectProvidedService(SPOnlineConnection connection)
         {
+            if (connection == null)
+                return false;
             connection.AccessToken = string.Empty;
             Environment.SetEnvironmentVariable("PNPPSHOST", string.Empty);
             Environment.SetEnvironmentVariable("PNPPSSITE", string.Empty);
-            if (connection == null)
-                return false;
             connection.Context = null;
             connection = null;
             return true;
         }
 
         internal static bool DisconnectCurrentService()
         {
+            if (SPOnlineConnection.CurrentConnection == null)
+                return false;
             SPOnlineConnection.CurrentConnection.AccessToken = string.Empty;
```

Comment:
As this line dereferences "SPOnlineConnection.CurrentConnection", the null check has to be before it